### PR TITLE
[backport-release-0.4] dont use hard-coded value for manager namespace

### PR DESCRIPTION
**Pull request by nader-ziada**
**What type of PR is this?**
/kind bug

**What this PR does / why we need it**:
backport of PR #1208 

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

_Please confirm that if this PR changes any image versions, then that's the sole change this PR makes._

**TODOs**:

- [ ] squashed commits
- [ ] includes documentation
- [ ] adds unit tests

**Release note**:

```release-note
Remove hardcoded namespace value
```

### CODE REVIEW

File: util/system/namespace.go [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+import "os"
+
+const (
+	// NamespaceEnvVarName is the env var coming from DownwardAPI in the manager manifest
+	NamespaceEnvVarName = "POD_NAMESPACE"
+	// DefaultNamespace is the default value from manifest
+	DefaultNamespace = "capz-system"
+)
+
+// GetManagerNamespace return the namespace where the controller is running
```
Comment (shysank) [inline util/system/namespace.go:28]:
```suggestion
// GetManagerNamespace returns the namespace where the controller is running.
```

**Review by shysank — APPROVED** [review]
/lgtm with small nit

**Review by devigned — APPROVED** [review]
/lgtm

**Review by CecileRobertMichon — APPROVED** [review]
/approve
/retest

**Pull request merged**